src/main/java/com/okta/sdk/clients/UserApiClient.java

```diff
@@ -296,6 +593,109 @@ public LoginCredentials changeRecoveryQuestion(String userId, String currentPass
+    }
+
+    /**
+     * Get all users up to a limit using cursor with paging info.
+     *
+     * @param  after               The cursor that determines which users to return after.
+     * @param  limit               The max number of results returned.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsAfterCursorWithLimit(String after, int limit) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseAfterCursorWithLimit(after, limit));
+    }
+
+    /**
+     * Get all users from url with paging info.
+     *
+     * @param  url                 The URL to get users from.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsByUrl(String url) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseByUrl(url));
+    }
+
+    /**
+     * Get all users using advanced search with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearch(FilterBuilder filterBuilder) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseWithAdvancedSearch(filterBuilder));
+    }
+
+    /**
+     * Get all users up to a limit using advanced search with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @param  limit               The max number of results returned.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearchAndLimit(FilterBuilder filterBuilder, int limit) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseWithAdvancedSearchAndLimit(filterBuilder, limit));
+    }
+
+    /**
+     * Get all users up to a limit using advanced search and cursor with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @param  limit               The max number of results returned.
+     * @param  after               The cursor that determines which users to return after.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearchAndLimitAndAfterCursor(FilterBuilder filterBuilder, int limit, String after) throws IOException {
```

Comment:
We've had the convention of calling the q parameter the search functionality (both in code and on developer.okta.com, see http://developer.okta.com/docs/api/resources/users.html#list-users-with-search) so I thought advanced search was the official term we used for this. I've also seen that in the SDK we're using the term query (```getUsersWithQuery```) more than search, so there's already a base for confusion when reading the SDK and the documentation on developer.okta.com. I thought that it would be a good idea to be consistent with the docs now that we have the chance, instead of making it confusing saying that the new functionality is called search but it's not the search that we document online, it's what we call advanced search online.

Regarding the other part of the name for this method, I agree it is very verbose; I was first thinking of using the name ```getUsersPagedResultsWithAdvancedSearch``` and overload this method with all 3 parameter combinations (search, search + limit, search + limit + cursor). I didn't end up doing that to follow the convention we use with filtering, which is spelling out the parameters in the name of the function. So I followed consistency in this case.

Let me know what you think we should call it; I think we should get rid of parameter names in the method name, and instead assume they read the javadocs and understands. I would then call them all ```getUsersPagedResultsWithAdvancedSearch(...)```.